Src/WitsmlExplorer.Frontend/components/ContextMenus/MessageObjectContextMenu.tsx

```diff
@@ -0,0 +1,96 @@
+import React from "react";
+import { DisplayModalAction, HideModalAction, HideContextMenuAction } from "../../contexts/operationStateReducer";
+import OperationType from "../../contexts/operationType";
+import { ListItemIcon, MenuItem, Typography } from "@material-ui/core";
+import ContextMenu from "./ContextMenu";
+import JobService, { JobType } from "../../services/jobService";
+import { Server } from "../../models/server";
+import ConfirmModal from "../Modals/ConfirmModal";
+import { MessageObjectRow } from "../ContentViews/MessagesListView";
+import { DeleteIcon, FormatLineSpacingIcon } from "../Icons";
+import ModificationType from "../../contexts/modificationType";
+import MessageObjectService from "../../services/messageObjectService";
+import { UpdateWellboreMessageAction, UpdateWellboreMessagesAction } from "../../contexts/navigationStateReducer";
+import MessagePropertiesModal, { MessagePropertiesModalProps } from "../Modals/MessagePropertiesModal";
+import { PropertiesModalMode } from "../Modals/ModalParts";
+
+export interface MessageObjectContextMenuProps {
+  checkedMessageObjectRows: MessageObjectRow[];
+  dispatchOperation: (action: DisplayModalAction | HideContextMenuAction | HideModalAction) => void;
+  dispatchNavigation: (action: UpdateWellboreMessagesAction | UpdateWellboreMessageAction) => void;
+  servers: Server[];
+  selectedServer: Server;
+}
+
+const MessageObjectContextMenu = (props: MessageObjectContextMenuProps): React.ReactElement => {
+  const { checkedMessageObjectRows, dispatchOperation, dispatchNavigation } = props;
+
+  const deleteMessageObjects = async () => {
+    dispatchOperation({ type: OperationType.HideModal });
+    const job = {
+      messageObjects: checkedMessageObjectRows.map((row) => ({
+        wellUid: row.wellUid,
+        wellboreUid: row.wellboreUid,
+        uid: row.uid
+      }))
+    };
+
+    await JobService.orderJob(JobType.DeleteMessageObjects, job);
+    const freshMessages = await MessageObjectService.getMessages(job.messageObjects[0].wellUid, job.messageObjects[0].wellboreUid);
```

Comment:
Will this actually return the correct list? Does the server have enough time to delete messages before this code runs and retrieves messages again?

And, is it necessary? I assume the DeleteMessageObjectsWorker returns a RefreshWellbore result and thus the wellbore should be refreshed?

Reply:
Thanks for feedback
I was thinking the same thing with refreshwellbore, but it did not seem like the list was refreshed afterwards.
It might be that the list was big (900+ msgs) and that a delay was the reason.

Anyways I will check. It might also be the case that the post-fetch runs parallel to the delete in this case.

Reply:
Look into `refreshHandler` you might want to add logic there to handle refreshing messages

Reply:
Thanks! 
I moved the refresh logic to the refresh handler, this made everything much cleaner and responsive.